GitUI/UserControls/RevisionGrid/Graph/RevisionGraphRevision.cs

```diff
@@ -92,7 +93,7 @@ public int EnsureScoreIsAbove(int minimalScore)
 
         public ImmutableStack<RevisionGraphRevision> Parents => _parents;
         public ImmutableStack<RevisionGraphRevision> Children => _children;
-        public SynchronizedCollection<RevisionGraphSegment> StartSegments { get; }
+        public ConcurrentBag<RevisionGraphSegment> StartSegments { get; }
```

Comment:
💭 I believe `ConcurrentBag` is unordered. Is that going to cause a problem here?

Reply:
I haven't observed any side-effects. 
We don't appear to be relying on order of items in all but one case:
https://github.com/gitextensions/gitextensions/blob/f840a0ab83a461035f816951fe39c752a92abc4c/GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs#L314-L318

@mstv you've looked at this code recently. Any thought?